src/remote.ts

```diff
@@ -0,0 +1,145 @@
+import { ChildProcess } from 'child_process'
+import { Headers } from 'headers-utils'
+import { Serializable } from 'node:child_process'
+import {
+  createInterceptor,
+  InterceptorApi,
+  InterceptorOptions,
+  IsomorphicRequest,
+  Resolver,
+} from './createInterceptor'
+
+type ProcessEventListener = (message: Serializable, ...args: any[]) => void
+
+export type CreateRemoteInterceptorOptions = Omit<
+  InterceptorOptions,
+  'resolver'
+>
+
+export interface CreateRemoteResolverOptions {
+  process: ChildProcess
+  resolver: Resolver
+}
+
+function requestReviver(key: string, value: any) {
+  switch (key) {
+    case 'url':
+      return new URL(value)
+
+    case 'headers':
+      return new Headers(value)
+
+    default:
+      return value
+  }
+}
+
+/**
+ * Creates a remote request interceptor that delegates
+ * the mocked response resolution to the parent process.
+ * The parent process must establish a remote resolver
+ * by calling `createRemoteResolver` function.
+ */
+export function createRemoteInterceptor(
+  options: CreateRemoteInterceptorOptions
+): InterceptorApi {
+  if (!process.connected) {
+    throw new Error(
+      `Failed to create a remote interceptor: the current process (${process.pid}) does not have a parent. Please make sure you're spawning this process as a child process in order to use remote request interception.`
+    )
+  }
+
+  if (typeof process.send === 'undefined') {
+    throw new Error(
+      `\
+Failed to create a remote interceptor: the current process (${process.pid}) does not have the IPC enabled. Please make sure you're spawning this process with the "ipc" stdio value set:
+
+spawn('node', ['module.js'], { stdio: ['ipc'] })\
+`
+    )
+  }
+
+  let handleParentMessage: ProcessEventListener
+
+  const interceptor = createInterceptor({
+    ...options,
+    resolver(request) {
+      const serializedRequest = JSON.stringify(request)
+      process.send?.(`request:${serializedRequest}`)
```

Comment:
Should be able to drop this condition now that the error message is in place

```suggestion
      process.send(`request:${serializedRequest}`)
```

Reply:
Logically—yes, but TypeScript still treats this `send` as potentially undefined. I suppose TS treats this closure as a runtime expression that doesn't guarantee that `process.send` has been checked before (or it can be a TS bug). 

So we can either use optional chaining or add another `typeof process.send` check here (which is redundant). Do you know about another option we could try?